### ISSUE TITLE
VSCode Breakpoint not hitting/stopping when run in debugger mode - Golang

### ISSUE THREAD

**Issue by riderrocks** [golang/vscode-go#2417]
### What version of Go, VS Code & VS Code Go extension are you using?
<details><summary>Version Information</summary><br>

* Run `go version` to get version of Go from _the VS Code integrated terminal_.
	go version go1.19 darwin/amd64
* Run `gopls -v version` to get version of Gopls from _the VS Code integrated terminal_.
	 Unknown command: gopls
* Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders.
	1.70.2
      e4503b30fc78200f846c62cf8091b76ff5547662
       x64
* Check your installed extensions to get the version of the VS Code Go extension
	Go v0.35.2
* Run Ctrl+Shift+P (Cmd+Shift+P on Mac OS) > `Go: Locate Configured Go Tools` command.
	Checking configured tools....
GOBIN: undefined
toolsGopath: 
gopath: /Users/ananthraghav/go
GOROOT: /usr/local/go
PATH: /Users/ananthraghav/google-cloud-sdk/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/mongodb/bin/:/usr/local/go/bin

	go:	/usr/local/go/bin/go: go version go1.19 darwin/amd64

	gotests:	/Users/ananthraghav/go/bin/gotests	(version: v1.6.0 built with go: go1.19)
	gomodifytags:	/Users/ananthraghav/go/bin/gomodifytags	(version: v1.16.0 built with go: go1.19)
	impl:	/Users/ananthraghav/go/bin/impl	(version: v1.1.0 built with go: go1.19)
	goplay:	/Users/ananthraghav/go/bin/goplay	(version: v1.0.0 built with go: go1.19)
	dlv:	/Users/ananthraghav/go/bin/dlv	(version: v1.9.0 built with go: go1.19)
	staticcheck:	/Users/ananthraghav/go/bin/staticcheck	(version: v0.3.3 built with go: go1.19)
	gopls:	/Users/ananthraghav/go/bin/gopls	(version: v0.9.4 built with go: go1.19)

go env
Workspace Folder (le-diy): /Users/ananthraghav/go/src/stash.bms.bz/bms/le-diy
	GO111MODULE=""
	GOARCH="amd64"
	GOBIN=""
	GOCACHE="/Users/ananthraghav/Library/Caches/go-build"
	GOENV="/Users/ananthraghav/Library/Application Support/go/env"
	GOEXE=""
	GOEXPERIMENT=""
	GOFLAGS="-trimpath"
	GOHOSTARCH="amd64"
	GOHOSTOS="darwin"
	GOINSECURE=""
	GOMODCACHE="/Users/ananthraghav/go/pkg/mod"
	GONOPROXY=""
	GONOSUMDB=""
	GOOS="darwin"
	GOPATH="/Users/ananthraghav/go"
	GOPRIVATE=""
	GOPROXY="https://proxy.golang.org,direct"
	GOROOT="/usr/local/go"
	GOSUMDB="sum.golang.org"
	GOTMPDIR=""
	GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
	GOVCS=""
	GOVERSION="go1.19"
	GCCGO="gccgo"
	GOAMD64="v1"
	AR="ar"
	CC="clang"
	CXX="clang++"
	CGO_ENABLED="1"
	GOMOD="/Users/ananthraghav/go/src/stash.bms.bz/bms/le-diy/go.mod"
	GOWORK=""
	CGO_CFLAGS="-g -O2"
	CGO_CPPFLAGS=""
	CGO_CXXFLAGS="-g -O2"
	CGO_FFLAGS="-g -O2"
	CGO_LDFLAGS="-g -O2"
	PKG_CONFIG="pkg-config"
	GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/ym/b8142c6d0hlc3byfnh5bggmr0000gq/T/go-build37387940=/tmp/go-build -gno-record-gcc-switches -fno-common"
</details>

### Share the Go related settings you have added/edited

Run `Preferences: Open Settings (JSON)` command to open your settings.json file.
Share all the settings with the `go.` or `["go"]` or `gopls` prefixes.

### Describe the bug
When Start Debugging is fired, the program runs successfully. However the program doesn't stop at breakpoints.
Breakpoint is enabled and placed alright. But it doesn't hit

### Steps to reproduce the behavior:
1. Go to Run -> Start debugging
2. See in debug console - DAP server listening at: 127.0.0.1:50757
3. Process 31415 has exited with status 0
dlv dap (31383) exited with code: 0

### Screenshots or recordings
If applicable, add screenshots or recordings to help explain your problem.

<img width="1680" alt="Screenshot 2022-08-18 at 9 54 05 PM" src="https://user-images.githubusercontent.com/11162696/185445827-8378d907-e79a-429a-ba0b-e9d44821a544.png">

**Comment by suzmue** [comment]
@riderrocks Could you share the launch configuration you are using? Thanks!

**Comment by riderrocks** [comment]
@suzmue  Here is the launch.json
`{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${fileDirname}",
            "env": {},
            "args": []
        }
    ]
}`

**Comment by suzmue** [comment]
@riderrocks Thanks! Could you also enable logging and send the output from the debug console? I would guess that the breakpoint is not being set for some reason, but not sure why without more information.

You can enable logging by updating your launch configuration to include `"logOutput": "dap",  "showLog": true`:
```
 {
   "name": "Launch",
   "type": "go",
   "request": "launch",
   "mode": "auto",
   "program": "${fileDirname}",
   "logOutput": "dap",
   "showLog": true,
} 
```

**Comment by riderrocks** [comment]
<img width="1680" alt="Screenshot 2022-08-19 at 12 30 57 AM" src="https://user-images.githubusercontent.com/11162696/185473630-274eb630-cc0c-470d-8745-669ff896d344.png">

Hi @suzmue 
There is no log apperaring on debug console.

For logOutput: "dap"
`
Value is not accepted. Valid values: "debugger", "gdbwire", "lldbout", "debuglineerr", "rpc".(1)
`

`
 "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${fileDirname}",
            "logOutput": "dap",
            "showLog": true,
         } 
    ]
`

**Comment by suzmue** [comment]
Hmmm I'm not seeing any breakpoints requests.n Would you mind double checking that you have breakpoints set in the VS Code UI by going to the breakpoints view?

You can get there using the command prompt and running `> Run and Debug: Focus on Breakpoints View`:

![Screen Shot 2022-08-18 at 3 20 42 PM](https://user-images.githubusercontent.com/6634754/185476922-750ff5ef-a3c3-4c45-ba45-7f0e360b8f6e.png)
i

**Comment by riderrocks** [comment]
@suzmue Thanks! for the revert.

I have the set the breakpoints as can be seen in the image. It doesn't stop at the breakpoint instead prints the output and exits

<img width="1680" alt="Screenshot 2022-08-19 at 1 02 44 AM" src="https://user-images.githubusercontent.com/11162696/185479022-250e04ad-2eaf-4bc6-ae47-ab02f818dcc0.png">

**Comment by suzmue** [comment]
I see from your logs that noDebug is true, this means that it will not stop on the breakpoints.

Can you try running using the command `> Debug: Start Debugging`?

If you are using a keyboard short cut, make sure that the short cut you are using doesn't run the `Debug: Start without Debugging` command.

**Comment by riderrocks** [comment]
@suzmue I tried `Debug: Start Debugging `

But still the result is same that the breakpoints are not hit. Here is the screenshot and output.

<img width="1680" alt="Screenshot 2022-08-19 at 10 45 17 PM" src="https://user-images.githubusercontent.com/11162696/185672953-a85bc3f2-0ee4-4338-aa60-5bb11974bb96.png">
[output1908.txt](https://github.com/golang/vscode-go/files/9384112/output1908.txt)

**Comment by suzmue** [comment]
I see that is now sending the setBreakpoints requests so that is progress!

The error that it is getting is "could not find file /Users/ananthraghav/Desktop/Test/main.go".  Is this project by chance in a symlinked directory?

If not, can you try setting `"stopOnEntry": true` in your launch configuration. This should allow us to stop in the debug session without having to set a breakpoint. Once the debug session is running can you then type `dlv sources` in the debug console and look at the list of sources. You should be able to find the main.go file in that list somewhere.

**Comment by riderrocks** [comment]
Hi @suzmue  The file is not a symbolic link. 
`-rw-r--r--  1 ananthraghav  staff  177 20 Aug 11:54 main.go`

Tried the above suggestion. The main.go exists on the list. However the debugger doesn't allow the placement of breakpoint.
<img width="1680" alt="Screenshot 2022-08-20 at 11 54 48 AM" src="https://user-images.githubusercontent.com/11162696/185732084-80f06833-44fd-49ed-aa19-582624087b4b.png">

**Comment by suzmue** [comment]
Was the path of main.go in the list of sources the same as `/Users/ananthraghav/Desktop/Test/main.go`?

If not, add this to your launch configuration with the directory from the list of sources:
```
"substitutePath": [{"from": "/Users/ananthraghav/Desktop/Test", "to": directoryFromSources}]
```

Otherwise lmk if the paths are identical (with a screenshot ideally :) )

**Comment by riderrocks** [comment]
Hi @suzmue 
I put the substitutePath in launch.json and now breakpoints are hit :) 
Thanks much!

I wanted to know without mentioning the substitute path how can the same result be achieved. 

<img width="1680" alt="Screenshot 2022-08-27 at 12 15 05 PM" src="https://user-images.githubusercontent.com/11162696/187018604-2286fb53-9eb5-4ed1-924c-7c9714e06658.png">
 

<img width="1680" alt="Screenshot 2022-08-27 at 12 14 26 PM" src="https://user-images.githubusercontent.com/11162696/187018600-7c86c280-a6df-4c25-ba7a-d6b2e89f4947.png">

**Comment by cwh12355** [comment]
I appear same issue,I try the up method,not stop running，**still not resolved**
this is log:
``` 
Starting: /home/cwh/Go/bin/dlv dap --only-same-user=false --log=true --log-output=dap --listen=127.0.0.1:3174 --log-dest=3 from /home/cwh/Go/src/numb
DAP server listening at: 127.0.0.1:3174
2023-03-24T17:14:58+08:00 debug layer=dap DAP server pid = 14472
2023-03-24T17:14:58+08:00 debug layer=dap DAP connection 1 started
2023-03-24T17:14:58+08:00 debug layer=dap [<- from client]{"seq":1,"type":"request","command":"initialize","arguments":{"clientID":"vscode","clientName":"Visual Studio Code","adapterID":"go","locale":"zh-cn","linesStartAt1":true,"columnsStartAt1":true,"pathFormat":"path","supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"supportsInvalidatedEvent":true,"supportsMemoryEvent":true}}
2023-03-24T17:14:58+08:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"supportsSetVariable":true,"supportsExceptionInfoRequest":true,"supportsDelayedStackTraceLoading":true,"supportsLogPoints":true,"supportsDisassembleRequest":true,"supportsClipboardContext":true,"supportsSteppingGranularity":true,"supportsInstructionBreakpoints":true}}
2023-03-24T17:14:58+08:00 debug layer=dap [<- from client]{"seq":2,"type":"request","command":"launch","arguments":{"name":"Launch Package","type":"go","request":"launch","mode":"debug","program":"./numb.go","dlvFlags":["--only-same-user=false"],"stopOnEntry":true,"logOutput":"dap","showLog":true,"substitutePath":[{"from":"/home/cwh/Go/src/numb/","to":"./test"}],"__configurationTarget":6,"packagePathToGoModPathMap":{"/home/cwh/Go/src/numb":"/home/cwh/Go/src/numb"},"debugAdapter":"dlv-dap","showRegisters":false,"showGlobalVariables":false,"hideSystemGoroutines":false,"dlvToolPath":"/home/cwh/Go/bin/dlv","env":{"GO111MODULE":"on","GOPATH":"/home/cwh/Go"},"__buildDir":"/home/cwh/Go/src/numb","__sessionId":"b5ee25aa-fb8e-4044-a2ac-9546a33749bd"}}
2023-03-24T17:14:58+08:00 debug layer=dap parsed launch config: {
	"mode": "debug",
	"program": "./numb.go",
	"env": {
		"GO111MODULE": "on",
		"GOPATH": "/home/cwh/Go"
	},
	"stopOnEntry": true,
	"backend": "default",
	"stackTraceDepth": 50,
	"substitutePath": [
		{
			"from": "/home/cwh/Go/src/numb/",
			"to": "./test"
		}
	]
}
2023-03-24T17:14:59+08:00 debug layer=dap building from "/home/cwh/Go/src/numb": [go build -o /home/cwh/Go/src/numb/__debug_bin -gcflags all=-N -l ./numb.go]
2023-03-24T17:14:59+08:00 debug layer=dap launching binary '/home/cwh/Go/src/numb/__debug_bin' with config: {
	"mode": "debug",
	"program": "/home/cwh/Go/src/numb/numb.go",
	"cwd": "/home/cwh/Go/src/numb",
	"output": "/home/cwh/Go/src/numb/__debug_bin",
	"dlvCwd": "/home/cwh/Go/src/numb",
	"env": {
		"GO111MODULE": "on",
		"GOPATH": "/home/cwh/Go"
	},
	"stopOnEntry": true,
	"backend": "default",
	"stackTraceDepth": 50,
	"substitutePath": [
		{
			"from": "/home/cwh/Go/src/numb/",
			"to": "./test"
		}
	]
}
```
@suzmue

**Comment by wisefool01** [comment]
> 

Thanks. I forgot that my root directory was a symlink which was causing this issue. I think this substitution will solve some of the other issues I've been facing too.

**Comment by bitwombat** [comment]
@cwh12355 please open a new issue rather than tacking onto this one - easier for the people providing support.

@suzmue These are the most kick-ass troubleshooting steps. A model for how it should be done! Thanks.

**Comment by maximLyakhov** [comment]
Stops on breakpoint with such config:
`{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Launch",
      "type": "go",
      "request": "launch",
      "program": "./"
    }
  ]
}
`
In the root of a project lies entry `main.go` file.

**Comment by nisimpson** [comment]
> > "substitutePath": [{"from": "/Users/ananthraghav/Desktop/Test", "to": directoryFromSources}]
> 
> Thanks. I forgot that my root directory was a symlink which was causing this issue. I think this substitution will solve some of the other issues I've been facing too.

This fixed the issue for me as well. Note that you only need to substitute up to the common path:

```
// launch.json
// Scenario:
//     Project: /workspaces/nisimpson/myproject
//     Symlink: /home/nisimpson/w/myproject
{
  "configurations": [
    {
       "substitutePath": [{
          "from": "/home/nisimpson/w",
          "to": "/workspaces/nisimpson"
        }],
        ...
    }
  ]
}
```

**Comment by klm127** [comment]
I fixed my issue by ensuring breakpoints were on a line that had code. They seem to be skipped if they are on a line that is just whitespace.

**Comment by ghasemloo** [comment]
This solved the issue for me as well.

TLDR:
if folder you are using with VSCode has a symlink in its path,
you need to tell VSCode to replace the symlink with real ones,
else it would not find the files (doesn't work with symlinks).

Not clear why it doesn't work with symlinks.

in your `launch.json` add:
```javascript
{
  "configurations": [
    {
      "substitutePath": [{
        // location of project folder in VSCode: 
        //   "/path/to/symlink-parent/myproject"
        // location of symlink parent:
        //   "/path/to/symlink-parent"
        // location of real parent:            
        //   "/path/to/real-parent"
        // location of real project folder:
        //   "/path/to/real-parent/myproject"
        //
        // map the symlink parent to real parent.
        //
          "from": "/path/to/symlink-parent",
          "to":     "/path/to/real-parent"
        }],
    }
  ]
}
```

**Comment by RafalSkolasinski** [comment]
Not handling symlinks is indeed quite annoying. I had my project's folder symlinked and was also puzzled why debugging was not working.

**Comment by McNull** [comment]
I wish I had found this discussion earlier -- was about to reinstall my operating system.

**Comment by Vadim-Maklakov** [comment]
Hi all, I have same problem with hitting/ stopping  programm on the break point on the Linux Host (Debian 12).
VSC and Go version  on problem host 
`go version go1.24.3 linux/amd64
Delve Debugger Version: 1.25.0
Build: $Id: e323af07680631b9cbdcc9d02e0a37891d12d972      
VCS Version: 1.101.2
Commit: 2901c5ac6db8a986a5666c3af51ff804d05af0d4                                                                 
Date: 2025-06-24T20:27:15.391Z
Electron: 35.5.1
ElectronBuildId: 11727614
Chromium: 134.0.6998.205
Node.js: 22.15.1
V8: 13.4.114.21-electron.0
OS: Linux x64 6.1.0-17-amd64 - Debian 12`
VSC on Windows 10 host has  same version  1.101.2, works  correctly  and stopping/hitting  in the conditional break point how and  the LiteIDE X 38.3
Simple programm for check correct work of  stopping/hitting breakpoint for debbuger main.go

> package main
> 
> import "fmt"
> 
> func main() {
> 	PrintHello()
> 	for i := 0; i < 5; i++ {
> 		PrintHello()
> 		PrintNumber(i) // Here conditional breakpoint with i == 2
> 	}
> }
> 
> // PrintHello print Hello, Go
> func PrintHello() {
> 	fmt.Println("Hello, Go")
> }
> 
> // PrintNumber writes a number using the fmt.Println function
> func PrintNumber(number int) {
> 	fmt.Println(number)
> }
> 

launch.json config
`{
    "configurations": [

    {
        "name": "Launch Package",
        "type": "go",
        "request": "launch",
        "mode": "auto",
        "program": "${workspaceFolder}",
        "logOutput": "dap",
        "showLog": true,
    }
    ]
}`
debug log - debugger see breakpoint and it's type - conditional, but doesn't hitting/stopping
`Starting: /home/mvg/go/bin/dlv dap --log=true --log-output=dap --listen=127.0.0.1:41397 --log-dest=3 from /home/mvg/Documents/Go/02_ProGo/03 - Using the Go Tools/tools
DAP server listening at: 127.0.0.1:41397
2025-07-04T23:24:09+03:00 debug layer=dap DAP server pid = 73225
2025-07-04T23:24:09+03:00 debug layer=dap DAP connection 1 started
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":1,"type":"request","command":"initialize","arguments":{"clientID":"vscode","clientName":"Visual Studio Code","adapterID":"go","locale":"en","linesStartAt1":true,"columnsStartAt1":true,"pathFormat":"path","supportsVariableType":true,"supportsVariablePaging":true,"supportsRunInTerminalRequest":true,"supportsMemoryReferences":true,"supportsProgressReporting":true,"supportsInvalidatedEvent":true,"supportsMemoryEvent":true,"supportsArgsCanBeInterpretedByShell":true,"supportsStartDebuggingRequest":true}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":1,"success":true,"command":"initialize","body":{"supportsConfigurationDoneRequest":true,"supportsFunctionBreakpoints":true,"supportsConditionalBreakpoints":true,"supportsEvaluateForHovers":true,"supportsSetVariable":true,"supportsExceptionInfoRequest":true,"supportsDelayedStackTraceLoading":true,"supportsLogPoints":true,"supportsDisassembleRequest":true,"supportsClipboardContext":true,"supportsSteppingGranularity":true,"supportsInstructionBreakpoints":true}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":2,"type":"request","command":"launch","arguments":{"name":"Launch Package","type":"go","request":"launch","mode":"debug","program":".","logOutput":"dap","showLog":true,"__configurationTarget":6,"packagePathToGoModPathMap":{},"debugAdapter":"dlv-dap","showRegisters":false,"showGlobalVariables":false,"substitutePath":[],"dlvFlags":[],"hideSystemGoroutines":false,"dlvToolPath":"/home/mvg/go/bin/dlv","env":{"GOPATH":"/home/mvg/go"},"__buildDir":"/home/mvg/Documents/Go/02_ProGo/03 - Using the Go Tools/tools","__sessionId":"9d95ae0d-d634-4c77-8e2e-f456fcd8aeec"}}
2025-07-04T23:24:09+03:00 debug layer=dap parsed launch config: {
	"mode": "debug",
	"program": ".",
	"buildFlags": {},
	"env": {
		"GOPATH": "/home/mvg/go"
	},
	"backend": "default",
	"stackTraceDepth": 50
}
2025-07-04T23:24:09+03:00 debug layer=dap building from "/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools": [go build -o /wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/__debug_bin62806632 -gcflags all=-N -l .]
2025-07-04T23:24:09+03:00 debug layer=dap launching binary '/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/__debug_bin62806632' with config: {
	"mode": "debug",
	"program": "/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools",
	"cwd": "/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools",
	"buildFlags": {},
	"output": "/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/__debug_bin62806632",
	"dlvCwd": "/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools",
	"env": {
		"GOPATH": "/home/mvg/go"
	},
	"backend": "default",
	"stackTraceDepth": 50
}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"process","body":{"name":"\"/wd1/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/__debug_bin62806632\"","systemProcessId":73333,"isLocalProcess":true,"startMethod":"launch"}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"initialized"}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":2,"success":true,"command":"launch"}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":3,"type":"request","command":"setBreakpoints","arguments":{"source":{"name":"main.go","path":"/home/mvg/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/main.go"},**"breakpoints":[{"line":15,"condition":"i == 2"}],"lines":[15]}}**
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":3,"success":true,"command":"setBreakpoints","body":{"breakpoints":[{"verified":false,"message":"could not find file /home/mvg/Documents/Go/02_ProGo/03 - Using the Go Tools/tools/main.go"}]}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":4,"type":"request","command":"setFunctionBreakpoints","arguments":{"breakpoints":[]}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":4,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":5,"type":"request","command":"setInstructionBreakpoints","arguments":{"breakpoints":[]}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":5,"success":true,"command":"setInstructionBreakpoints","body":{"breakpoints":[]}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":6,"type":"request","command":"configurationDone"}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"output","body":{"category":"console","output":"Type 'dlv help' for list of commands.\n"}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":6,"success":true,"command":"configurationDone"}
Type 'dlv help' for list of commands. - **Programm finished, doesn't hitting or stopint in the breakpoint**
Hello, Go
Hello, Go
0
Hello, Go
1
Hello, Go
2
Hello, Go
3
Hello, Go
4
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"terminated","body":{}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":7,"type":"request","command":"threads"}
2025-07-04T23:24:09+03:00 debug layer=dap Process 73333 has exited with status 0
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":7,"success":true,"command":"threads","body":{"threads":[{"id":1,"name":"Dummy"}]}}
2025-07-04T23:24:09+03:00 debug layer=dap [<- from client]{"seq":8,"type":"request","command":"disconnect","arguments":{}}
2025-07-04T23:24:09+03:00 debug layer=dap halting
2025-07-04T23:24:09+03:00 debug layer=dap process not running
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"output","body":{"category":"console","output":"Process 73333 has exited with status 0\n"}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"output","body":{"category":"console","output":"Detaching\n"}}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":8,"success":true,"command":"disconnect"}
2025-07-04T23:24:09+03:00 debug layer=dap [-> to client]{"seq":0,"type":"event","event":"terminated","body":{}}
Process 73333 has exited with status 0
Detaching`

It's kind of a disgrace   ©The dog's heart 1988

**Comment by Vadim-Maklakov** [comment]
No news is good news © Old Englist idiom))
 It's funny that a super puper mega corporations with a turnover of hundreds of billions of dollars can't fix some minor bug for three years...

**Comment by firelizzard18** [comment]
Go isn't exactly a priority for google. Specifically, the executives who make hiring/resource allocation decisions likely don't care whether you're having a good experience. I'm sure the Go team would love to have two or three times as many developers working on vscode-go, but unless google/alphabet decides they care we have to deal with the world as it is.

**Comment by bitwombat** [comment]
I'm worried Carbon and Dart are siphoning off enthusiasm/energy.

**Comment by mpictor** [comment]
> I wish I had found this discussion earlier -- was about to reinstall my operating system.

💯 :100: :100: :100:

Now that is pain I can relate to!
